### ISSUE TITLE
CRD Generator is not generating multiple custom resource versions in case one version is tied to a reconciler

### ISSUE THREAD

**Issue by moayad-alyaghshi** [quarkiverse/quarkus-operator-sdk#886]
Hi,

Fabric8 CRD generator supports multiple versions of the same kind:
https://github.com/fabric8io/kubernetes-client/blob/main/doc/CRD-generator.md#multiple-custom-resource-versions

This does not work for us. We have two versions (v1 and v2) of the same custom resource. We set [quarkus.operator-sdk.crd.generate-all](https://docs.quarkiverse.io/quarkus-operator-sdk/dev/includes/quarkus-operator-sdk.html#quarkus-operator-sdk_quarkus-operator-sdk-crd-generate-all) to true. When no version is tied to a reconciler, both versions will be generated, but when we use one version (v2) by the reconciler, only this version is generated and the other (v1) is ignored.

Is this intended behavior or a bug?

note: Quarkus platform version is 3.10.2

**Comment by metacosm** [comment]
Sorry for getting back to you so late, could you elaborate on your use case a little as I'm not sure I understand correctly what you're trying to accomplish?

**Comment by moayad-alyaghshi** [comment]
Hi,

We had in the beginning a custom resource with v1. This version was the only one available and it was tied to a reconciler, so it was auto generated from the java class using Fabric8 CRD generator and everything was working fine. Now the schema is changed, so we want to create v2, and we want the auto generated CRD to have both versions (v2 is stored, and v1 is only served). 

Generating multiple custom resource versions is supported by Fabric8 CRD generator, which is mentioned [here](https://github.com/fabric8io/kubernetes-client/blob/main/doc/CRD-generator.md#multiple-custom-resource-versions). We created another java class for the custom resource, annotated it with v2, and tied to the reconciler, but the result was not expected. This is the observed behavior:
- if v2 is the only one tied to a reconciler and no additional properties is set, only v2 is generated
- if v2 is the only one tied to a reconcile and [quarkus.operator-sdk.crd.generate-all](https://docs.quarkiverse.io/quarkus-operator-sdk/dev/includes/quarkus-operator-sdk.html#quarkus-operator-sdk_quarkus-operator-sdk-crd-generate-all) is set to true, only v2 is generated
- if no versions is the tied to a reconcile and [quarkus.operator-sdk.crd.generate-all](https://docs.quarkiverse.io/quarkus-operator-sdk/dev/includes/quarkus-operator-sdk.html#quarkus-operator-sdk_quarkus-operator-sdk-crd-generate-all) is set to true, both versions are generated

The desired behavior that v2 is tied to a reconciler and both versions are generated, but we couldn't achieve that. Keep in mind that the issue here is old and the used Quarkus version is old (3.10.2), so not sure if anything changed since then.

I hope this explanation helps. I can provide you with a simple example in cased it is needed.
Thanks.

**Comment by metacosm** [comment]
Hi @moayad-alyaghshi, ok, this is more or less what I understood but the clarification helps! Will look into it.

**Comment by metacosm** [comment]
The problem is that the generated CRD name doesn't contain the CR version, which means that currently, only one version can be generated.

**Comment by metacosm** [comment]
I've opened https://github.com/quarkiverse/quarkus-operator-sdk/pull/971 which shows the current behavior.

**Comment by moayad-alyaghshi** [comment]
@metacosm 
We just want to generate the two versions of the CR. we looked into **quarkus.operator-sdk.crd.generate-all** since we couldn't achieve that. 

Attached is a simple example. There is a CR with two versions (only v2 is tied to a reconciler). We want both versions to be generated into the CRD.

[code-with-quarkus.zip](https://github.com/user-attachments/files/17276241/code-with-quarkus.zip)

**Comment by metacosm** [comment]
Hi @moayad-alyaghshi,
There was indeed an issue which will be addressed in 6.8.3. However, I don't think it will be possible to generate both versions in separate files. They will be put in the same CRD. Would that be acceptable?

**Comment by metacosm** [comment]
Let me know if the associated PR solves your issue.

**Comment by moayad-alyaghshi** [comment]
Yes, generating the versions in the same CRD file is acceptable. 
Thank you
